### PR TITLE
110 add findtasks command

### PR DESCRIPTION
**Pull request by gavin331**
The 'findtasks' feature works similarly to the find command for the employees. The UG and DG have also been updated to include the 'findtasks' feature. Also updated the UI to show the employee/task panels whenever we run find or findtasks. Lastly, some minor refactoring was done.

The testcases for 'findtasks' have not been implemented yet.

### CODE REVIEW

**Review by BryanL2303 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/FindTasksCommand.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " project";
+
+    private final TaskNameContainsKeywordsPredicate predicate;
+
+    public FindTasksCommand(TaskNameContainsKeywordsPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
```
Comment (johnyoozhengxian) [inline src/main/java/seedu/address/logic/commands/FindTasksCommand.java:29]: LGTM

File: src/main/java/seedu/address/logic/parser/FindTasksCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Parses input arguments and creates a new FindTaskCommand object
+ */
+public class FindTasksCommandParser implements Parser<FindTasksCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindTasksCommand parse(String args) throws ParseException {
```
Comment (johnyoozhengxian) [inline src/main/java/seedu/address/logic/parser/FindTasksCommandParser.java:20]: LGTM

File: src/main/java/seedu/address/logic/commands/FindTasksCommand.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        if (!(other instanceof FindTasksCommand)) {
+            return false;
+        }
+
+        FindTasksCommand otherFindCommand = (FindTasksCommand) other;
+        return predicate.equals(otherFindCommand.predicate);
+    }
+
+    @Override
+    public String toString() {
```
Comment (johnyoozhengxian) [inline src/main/java/seedu/address/logic/commands/FindTasksCommand.java:54]: LGTM

File: src/main/java/seedu/address/logic/commands/FindTasksCommand.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof FindTasksCommand)) {
```
Comment (johnyoozhengxian) [inline src/main/java/seedu/address/logic/commands/FindTasksCommand.java:45]: LGTM

File: src/main/java/seedu/address/logic/commands/FindTasksCommand.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredTaskList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_TASKS_LISTED_OVERVIEW, model.getFilteredTaskList().size()),
+                false, false, true, false);
+    }
+
+    @Override
```
Comment (johnyoozhengxian) [inline src/main/java/seedu/address/logic/commands/FindTasksCommand.java:38]: LGTM

**Review by johnyoozhengxian — COMMENTED** [review]
LGTM maybe can add in some test cases

**Pull request merged**